addons/html_builder/static/tests/helpers.js

```diff
@@ -322,20 +335,151 @@ export async function waitForSnippetDialog() {
+ * @param {string} [options.groupName="general"] - The snippet group/category
+ * @param {number} [options.moduleId] - Module ID if snippet belongs to a module
+ * @param {string} [options.moduleDisplayName] - Human-readable module name
+ * @param {string} [options.content] - Static HTML content for the snippet
+ * @param {string[]} [options.keywords=[]] - Search keywords for the snippet
+ * @param {string} [options.imagePreview=""] - URL to preview image
+ * @param {boolean} [options.withName=false] - Whether to include snippet name in generated content
+ * @param {Function} [options.contentGenerator] - Function to generate dynamic content: (snippetName) => string
+ * @returns {Object} Snippet description object with standardized structure
+ */
+export function createSnippetDescription({
+    name,
+    groupName = "general",
+    moduleId,
+    moduleDisplayName,
+    content,
+    keywords = [],
+    imagePreview = "",
+    withName = false,
+    contentGenerator,
+}) {
+    let finalContent = content;
+
+    if (contentGenerator) {
+        finalContent = contentGenerator(withName ? name : "");
+    } else if (!content) {
+        finalContent = `<div>${name} Content</div>`;
+    }
+
+    const snippet = {
+        name,
+        groupName,
+        content: finalContent,
+        keywords,
+        imagePreview,
+    };
+
+    if (moduleId !== undefined) {
+        snippet.moduleId = moduleId;
+    }
+
+    if (moduleDisplayName !== undefined) {
+        snippet.moduleDisplayName = moduleDisplayName;
+    }
+
+    return snippet;
+}
+
+/**
+ * High-level helper to create multiple test snippets with consistent configuration
+ * Builds on createSnippetDescription to replace repetitive snippet functions in tests
+ * @param {Object} options - Configuration for snippet creation
+ * @param {Array} [options.snippets=[]] - Array of snippet configurations (see createSnippetDescription)
+ * @param {string} [options.snippets[].sectionContent] - Content for getBasicSection (alternative to content)
+ * @param {boolean} [options.useBasicSection=true] - Whether to use getBasicSection for content generation
+ * @param {boolean} [options.withName=false] - Whether to include snippet names in generated content
+ * @returns {Array} Array of snippet description objects created via createSnippetDescription
+ */
+export function createTestSnippets({ snippets = [], useBasicSection = true, withName = false }) {
+    return snippets.map((snippet) => {
+        const config = {
+            name: snippet.name,
+            groupName: snippet.groupName || "a",
+            keywords: snippet.keywords || [],
+            imagePreview: snippet.imagePreview || "",
+            withName: withName,
+        };
+
+        if (snippet.moduleId !== undefined) {
+            config.moduleId = snippet.moduleId;
+        }
+
+        if (snippet.moduleDisplayName !== undefined) {
+            config.moduleDisplayName = snippet.moduleDisplayName;
+        }
```

Comment:
Why do you need to check if !== undefined ? config.moduleId will be undefined in both case